### PR TITLE
Signing GHA images using cosign

### PR DESCRIPTION
**Pull request by javuto**
For context: https://github.blog/2021-12-06-safeguard-container-signing-capability-actions/

Adding `cosign` capabilities to test signing images.

### CODE REVIEW

File: .github/actions/build-publish-chainlink/action.yml [hunk 1]
```diff
@@ -127,3 +141,22 @@ runs:
...
+  - if: inputs.sign-images == 'true'
+    name: Install cosign
+    uses: sigstore/cosign-installer@1e95c1de343b5b0c23352d6417ee3e48d5bcd422 # v1.4.0
+    with:
+      cosign-release: 'v1.4.0'
+
+  - if: inputs.sign-images == 'true'
+    name: Write signing key to disk (only needed for `cosign sign --key`)
+    shell: sh
+    run: echo "${{ inputs.cosign-private-key }}" > cosign.key
```
Comment (HenryNguyen5) [inline .github/actions/build-publish-chainlink/action.yml:154]: Can we not pass this in stdin to `cosign sign`?

File: .github/actions/build-publish-chainlink/action.yml [hunk 2]
```diff
@@ -127,3 +141,22 @@ runs:
...
+
+  - if: inputs.sign-images == 'true'
+    name: Sign the published Docker image
+    shell: sh
+    env:
+      COSIGN_PASSWORD: "${{ inputs.cosign-password }}"
+    run: cosign sign --key cosign.key ${{ inputs.ecr-registry }}/${{ inputs.image-name }}@${{ steps.meta-nonroot.outputs.digest }}
```
Comment (HenryNguyen5) [inline .github/actions/build-publish-chainlink/action.yml:161]: Do we only want to sign the non-root digest here? Seems like we want to sign both generated images, no?

Comment (HenryNguyen5) [reply .github/actions/build-publish-chainlink/action.yml:161]: Also, the metadata action doesnt output the image digest, the build-push-action does. Another output the build-push-action provides is `metadata`, you might want to take a look in there and see if it makes forming the image uri easier

**Review by HenryNguyen5 — CHANGES_REQUESTED** [review]

File: .github/actions/build-publish-chainlink/action.yml (hunk 1 above)
Comment (javuto) [reply .github/actions/build-publish-chainlink/action.yml:154]: Unfortunately, `cosign` only reads the key from a file

**Review by HenryNguyen5 — APPROVED** [review]

**Review by HenryNguyen5 — APPROVED** [review]

**Review by HenryNguyen5 — APPROVED** [review]

**Pull request merged**